src/LiveComponent/src/EventListener/InterceptChildComponentRenderSubscriber.php

```diff
@@ -0,0 +1,113 @@
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\LiveComponent\Twig\DeterministicTwigIdCalculator;
+use Symfony\UX\LiveComponent\Util\FingerprintCalculator;
+use Symfony\UX\LiveComponent\Util\TwigAttributeHelper;
+use Symfony\UX\TwigComponent\ComponentFactory;
+use Symfony\UX\TwigComponent\ComponentStack;
+use Symfony\UX\TwigComponent\Event\PreCreateForRenderEvent;
+
+/**
+ * Responsible for rendering children as empty elements during a re-render.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class InterceptChildComponentRenderSubscriber implements EventSubscriberInterface
+{
+    public const CHILDREN_FINGERPRINTS_METADATA_KEY = 'children_fingerprints';
+
+    public function __construct(
+        private ComponentStack $componentStack,
+        private DeterministicTwigIdCalculator $deterministicTwigIdCalculator,
+        private FingerprintCalculator $fingerprintCalculator,
+        private TwigAttributeHelper $twigAttributeHelper,
+        private ComponentFactory $componentFactory,
+        private LiveComponentHydrator $liveComponentHydrator,
+    ) {
+    }
+
+    public function preComponentCreated(PreCreateForRenderEvent $event): void
+    {
+        // if there is already a component, that's a parent. Else, this is not a child.
+        if (!$this->componentStack->getCurrentComponent()) {
+            return;
+        }
+
+        $parentComponent = $this->componentStack->getCurrentComponent();
+        if (!$parentComponent->hasExtraMetadata(self::CHILDREN_FINGERPRINTS_METADATA_KEY)) {
+            return;
+        }
+
+        $childFingerprints = $parentComponent->getExtraMetadata(self::CHILDREN_FINGERPRINTS_METADATA_KEY);
+
+        // get the deterministic id for this child, but without incrementing the counter yet
+        $deterministicId = $this->deterministicTwigIdCalculator->calculateDeterministicId(increment: false);
+        if (!isset($childFingerprints[$deterministicId])) {
+            // child fingerprint wasn't set, it is likely a new child, allow it to render fully
+            return;
+        }
+
+        // increment the internal counter now to keep "counter" consistency if we're
+        // in a loop of children being rendered on the same line
+        // we need to do this because this component will *not* ever hit
+        // AddLiveAttributesSubscriber where the counter is normally incremented
+        $this->deterministicTwigIdCalculator->calculateDeterministicId(increment: true);
+
+        $newPropsFingerprint = $this->fingerprintCalculator->calculateFingerprint($event->getProps());
+
+        if ($childFingerprints[$deterministicId] === $newPropsFingerprint) {
+            // the props passed to create this child have *not* changed
+            // return an empty element so the frontend knows to keep the current child
+
+            $rendered = sprintf(
+                '<div data-live-id="%s"></div>',
```

Comment:
Will this be a problem if the child component isn't wrapped in a `div`?

Reply:
Excellent question! It should be ok. If the child element is actually a `span`, we actually account for this by correcting this `div` into a `span` right before we diff them (on the server, we have no idea what element to use, but once we're in JavaScript, we can find the child element in the current HTML and the "new" HTML via the `data-live-id`, then make them match).

The other potential weird thing (but JS tests confirm it's ok) is if the child element is, for example, a `<tr>`... where it's not legal to have a `<div>`. In other words, the new HTML would be `<table><div>...`. But, that would be wrong for just a moment (and only in memory, we haven't actually used the new HTML at this point) and then we would correct it to a `<tr>`.

So there *are* some gymnastics happening in the JS, but it seems to work.